### PR TITLE
BAEL-1085

### PR DESCRIPTION
**Pull request by Thoughtscript**

### CODE REVIEW

File: spring-rest/src/main/java/net/baeldung/ApplicationMain.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package net.baeldung;
```
Comment (pivovarit) [inline spring-rest/src/main/java/net/baeldung/ApplicationMain.java]: com.baeldung :)

File: spring-rest/src/main/java/net/baeldung/ApplicationMain.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.web.support.SpringBootServletInitializer;
+
+@SpringBootApplication
+public class ApplicationMain extends SpringBootServletInitializer {
```
Comment (pivovarit) [inline spring-rest/src/main/java/net/baeldung/ApplicationMain.java]: Do we need SpringBootServletInitializer? I believe that should work only with the @SpringBootApplication

File: spring-rest/src/main/java/net/baeldung/controllers/ExamplePostController.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import net.baeldung.transfer.LoginForm;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
+@RequestMapping("/post")
+public class ExamplePostController {
+
+    Logger log = LoggerFactory.getLogger(ExamplePostController.class);
```
Comment (pivovarit) [inline spring-rest/src/main/java/net/baeldung/controllers/ExamplePostController.java]: private static final log

File: spring-rest/src/main/java/net/baeldung/services/ExampleService.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package net.baeldung.services;
+
+import net.baeldung.transfer.LoginForm;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ExampleService {
+
+        public boolean fakeAuthenticate(LoginForm lf) {
+                return (lf.getUsername().equals(lf.getUsername()) && lf.getPassword() == lf.getPassword());
```
Comment (pivovarit) [inline spring-rest/src/main/java/net/baeldung/services/ExampleService.java]: Wouldn't it be easier to simply return true?

File: spring-rest/src/main/java/net/baeldung/transfer/LoginForm.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package net.baeldung.transfer;
+
+public class LoginForm {
+    private String username;
+    private String password;
+
+    public LoginForm() {};
```
Comment (pivovarit) [inline spring-rest/src/main/java/net/baeldung/transfer/LoginForm.java]: That's redundant

File: spring-security-mvc-socket/src/main/java/com/baeldung/springsecuredsockets/config/SecurityConfig.java [hunk 6]
```diff
@@ -89,7 +89,7 @@ protected void configure(HttpSecurity http) throws Exception {
                 .antMatchers("/", "/index", "/authenticate")
                 .permitAll()
                 .antMatchers("/secured/**/**",
-                        "/secured/success", "/secured/socket", "/secured/success")
+                        "/secured/success", "/secured/socket")
```
Comment (pivovarit) [inline spring-security-mvc-socket/src/main/java/com/baeldung/springsecuredsockets/config/SecurityConfig.java]: Is that related?

File: spring-rest/src/test/java/net/baeldung/controllers/ExamplePostControllerServiceUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                    .content(jsonBody)
+                    .contentType("application/json"))
+                    .andDo(print())
+                    .andExpect(status().isOk());
+        } catch (Exception e) {
+            System.out.println("Exception: " + e);
+        }
+    }
+
+    @After
```
Comment (pivovarit) [inline spring-rest/src/test/java/net/baeldung/controllers/ExamplePostControllerServiceUnitTest.java]: In @Before you are assigning something to mockMvc - so this @After does not make any impact and can be safely removed

File: spring-rest/src/test/java/net/baeldung/controllers/ExamplePostControllerIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+                    .andDo(print())
+                    .andExpect(status().isOk())
+                    .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
+                    .andExpect(jsonPath("$.text").value("Thanks For Posting!!!"));
+        } catch (Exception e) {
+            System.out.println("Exception: " + e);
+        }
+    }
+
+    @After
```
Comment (pivovarit) [inline spring-rest/src/test/java/net/baeldung/controllers/ExamplePostControllerIntegrationTest.java]: Same here, you can remove that

File: spring-rest/src/main/java/net/baeldung/ApplicationMain.java (hunk 1 above)
Comment (Thoughtscript) [inline spring-rest/src/main/java/net/baeldung/ApplicationMain.java]: Normally, the convention is com, org, then net (used rarely).  I thought it easier to simply and cleanly add my code into a separate grouping. However, I think I'm supposed to move this over to Spring Angular instead. So... my point is moot.

File: spring-rest/src/main/java/net/baeldung/controllers/ExamplePostController.java (hunk 3 above)
Comment (Thoughtscript) [inline spring-rest/src/main/java/net/baeldung/controllers/ExamplePostController.java]: Will make that change.

File: spring-rest/src/main/java/net/baeldung/services/ExampleService.java (hunk 4 above)
Comment (Thoughtscript) [inline spring-rest/src/main/java/net/baeldung/services/ExampleService.java]: Yeah - it could be simplified - just wanted to flesh that out - here it does qualify as a kind of authentication check since it compares values.

File: spring-security-mvc-socket/src/main/java/com/baeldung/springsecuredsockets/config/SecurityConfig.java (hunk 6 above)
Comment (Thoughtscript) [inline spring-security-mvc-socket/src/main/java/com/baeldung/springsecuredsockets/config/SecurityConfig.java]: No but this is a minor change requested elsewhere.

File: spring-rest/src/test/java/net/baeldung/controllers/ExamplePostControllerServiceUnitTest.java (hunk 7 above)
Comment (Thoughtscript) [inline spring-rest/src/test/java/net/baeldung/controllers/ExamplePostControllerServiceUnitTest.java]: True but leaving the full before, test, and after trio allows someone learning to see how a good test should be set up.

**Comment by Thoughtscript** [comment]
This code is being moved over to Spring-Rest-Angular per document review! Most of the points above are moot due to that change but some will be incorporated for sure. Thanks!

**Comment by Thoughtscript** [comment]
Had removed a duplicate ant matcher from BAEL-509 - this was the incorrect process - reverting now, article has been corrected, but will issue a separate PR for this
